### PR TITLE
Update Neptune dependency: `neptune-client` > `neptune`

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
The `neptune-client` Python package is deprecated in favor of `neptune`. I adjusted our Neptune integration to depend on the new package. Upon doing `zenml integration install neptune`, we also uninstall `neptune-client` first to prevent version conflicts.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/neptune/__init__.py [hunk 1]
```diff
@@ -29,9 +29,7 @@ class NeptuneIntegration(Integration):
     """Definition of the neptune.ai integration with ZenML."""
 
     NAME = NEPTUNE
-    REQUIREMENTS = [
-        "neptune-client>=0.16.10",
-    ]
+    REQUIREMENTS = ["neptune"]
```
Comment (strickvl) [inline src/zenml/integrations/neptune/__init__.py:32]: do we want / need to pin this to some lower version?

Comment (strickvl) [reply src/zenml/integrations/neptune/__init__.py:32]: Like even should it be >=1.0 at least?

**Review by strickvl — COMMENTED** [review]
LGTM, barring one question I had in the comments about pinning the version.

File: src/zenml/integrations/neptune/__init__.py (hunk 1 above)
Comment (fa9r) [reply src/zenml/integrations/neptune/__init__.py:32]: 1.0 is the smallest version that exists for `neptune` :)

Comment (strickvl) [reply src/zenml/integrations/neptune/__init__.py:32]: Would we then consider that a breaking change?

Comment (strickvl) [reply src/zenml/integrations/neptune/__init__.py:32]: Since people would be using <1.0 with `neptune-client`

**Review by strickvl — APPROVED** [review]

**Pull request merged**